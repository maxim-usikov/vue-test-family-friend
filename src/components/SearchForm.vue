<template>
  <form
    :class="$style.Form"
    @submit.prevent="handleSubmit"
  >
    <input
      v-model="query"
      :class="$style.Input"
      type="text"
      required
    >
    <button
      :class="$style.Button"
      type="submit"
    >
      send
    </button>
  </form>
</template>

<script>
// eslint-disable-next-line no-param-reassign, no-plusplus
const id = (i => () => i++)(1);

export default {
  name: 'SearchForm',

  data() {
    return {
      query: '',
    };
  },

  methods: {
    handleSubmit() {
      this.$emit('search', {
        id: id(),
        query: this.query,
        createdAt: new Date(),
      });

      this.query = '';
    },
  },
};
</script>

<style module>
.Form {
  display: flex;
}

.Input {
  border: 1px solid rgba(234, 234, 234, 1);
  border-right: none;
  outline: none;
  font-size: 1.4rem;
  padding: 4px 6px;
}

.Button {
  border: 1px solid rgba(76, 175, 80, 1);
  font-weight: bold;
  background: rgba(76, 175, 80, 1);
  outline: none;
  cursor: pointer;
}

.Button:hover {
  border-color: rgba(76, 175, 80, .8);
  background: rgba(76, 175, 80, .8);
}
</style>
