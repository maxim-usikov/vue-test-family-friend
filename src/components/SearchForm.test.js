import { shallowMount } from '@vue/test-utils';

import SearchForm from './SearchForm.vue';

describe('SearchForm', () => {
  let $c;

  beforeEach(() => {
    $c = shallowMount(SearchForm);
  });

  it('is a Vue instance', () => {
    expect($c.isVueInstance()).toBeTruthy();
  });

  it('name is `SearchForm`', () => {
    expect($c.name()).toBe('SearchForm');
  });

  it('renders correctly', () => {
    expect($c.element).toMatchSnapshot();
  });

  it('emits search', () => {
    const form = $c.find('form');
    $c.setData({ query: 'test' });

    form.trigger('submit');

    expect($c.emitted().search).toBeTruthy();
    expect($c.emitted().search.length).toBe(1);
    expect($c.emitted().search[0][0]).toMatchSnapshot({
      id: expect.any(Number),
      createdAt: expect.any(Date),
    });
  });
});
