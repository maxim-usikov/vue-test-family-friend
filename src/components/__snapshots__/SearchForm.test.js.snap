// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm emits search 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "query": "test",
}
`;

exports[`SearchForm renders correctly 1`] = `
<form
  class="Form"
>
  <input
    class="Input"
    required="required"
    type="text"
  />
   
  <button
    class="Button"
    type="submit"
  >
    
    send
  
  </button>
</form>
`;
